src/tiered/tiered_handle.c

```diff
@@ -400,6 +400,31 @@ __tiered_create_tier_tree(WT_SESSION_IMPL *session, WT_TIERED *tiered)
     return (ret);
 }
 
+/*
+ * __tiered_cleanup_tiers --
+ *     Cleanup and dereference all tiers.
+ */
+static void
+__tiered_cleanup_tiers(WT_SESSION_IMPL *session, WT_TIERED *tiered, bool final)
+{
+    uint32_t i;
+
+    for (i = 0; i < WT_TIERED_MAX_TIERS; i++) {
+        /*
+         * Do not use tier's dhandle if it's a final cleanup, it may be invalid.
+         * __wt_conn_dhandle_discard is responsible for the proper destruction of all the dhandles.
+         */
+        if (tiered->tiers[i].tier != NULL && !final) {
+            WT_ASSERT(session, tiered->tiers[i].tier->session_inuse > 0);
```

Comment:
This assert is now redundant, as it's done in `__wt_cursor_dhandle_decr_use`

Reply:
Done, thanks!